internal/pkg/fleetapi/action.go

```diff
@@ -348,6 +406,54 @@ func (a *ActionCancel) String() string {
 	return s.String()
 }
 
+func (a *ActionCancel) AckEvent() AckEvent {
+	return newAckEvent(a.ActionID, a.ActionType)
+}
+
+// ActionDiagnostics is a request to gather and upload a diagnostics bundle.
+type ActionDiagnostics struct {
+	ActionID   string `json:"action_id"`
+	ActionType string `json:"type"`
+	UploadID   string `json:"-"`
+	Err        error  `json:"-"`
+}
+
+// ID returns the ID of the action.
+func (a *ActionDiagnostics) ID() string {
+	return a.ActionID
+}
+
+// Type returns the type of the action.
+func (a *ActionDiagnostics) Type() string {
+	return a.ActionType
+}
+
+func (a *ActionDiagnostics) String() string {
+	var s strings.Builder
+	s.WriteString("action_id: ")
+	s.WriteString(a.ActionID)
+	s.WriteString(", type: ")
+	s.WriteString(a.ActionType)
+	return s.String()
+}
+
+func (a *ActionDiagnostics) AckEvent() AckEvent {
+	event := newAckEvent(a.ActionID, a.ActionType)
+	if a.Err != nil {
+		event.Error = a.Err.Error()
+	}
+	if a.UploadID != "" {
+		var payload struct {
+			UploadID string `json:"upload_id"`
+		}
+		payload.UploadID = a.UploadID
+		p, _ := json.Marshal(payload)
+		event.Payload = p
+	}
```

Comment:
Should we instead use `ActionResponse` to track the upload_id instead of `Payload`?
fleet-server already serializes this when it's handling events (we may also want to change the retry-mechanism to use this as well for consistency)

Reply:
Which one of these is written to `.fleet-actions-results`?
[Fleet UI](https://github.com/elastic/kibana/pull/142369/files#r1016816810) currently expects `data.file_id` in results docs. We have to align the implementation so that Fleet UI finds the uploaded file that corresponds to an action.

Reply:
OK, so the UI expects that the data attribute of the action result to be an object with `file_id`.
So Instead we should set the the `event.Data`, however it appears as if the agent does not send it as it's not defined in the event: https://github.com/elastic/elastic-agent/blob/fefe64fff4369bddd806fdccf691ef6213adab42/internal/pkg/fleetapi/ack_cmd.go#L23

however it's defined by the fleet-server https://github.com/elastic/fleet-server/blob/main/internal/pkg/api/schema.go#L166